File: app/src/components/MatchInfoCard.js
// Maç bilgi kartı (detay/Özet): bayrak + lig + lig haftası, tarih·gün·saat,
// hava, stadyum. Yalnız GERÇEK veri gösterilir; olmayan satır hiç çizilmez
// (uydurma yok). Veri kaynağı: backend m.info (+ m.league, m.date).
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { colors, spacing, radius, shadow } from '../theme';
import { countryFlag } from '../utils';

const WEEKDAY = ['Pazar', 'Pazartesi', 'Salı', 'Çarşamba', 'Perşembe', 'Cuma', 'Cumartesi'];
const AY = ['Ocak', 'Şubat', 'Mart', 'Nisan', 'Mayıs', 'Haziran', 'Temmuz', 'Ağustos', 'Eylül', 'Ekim', 'Kasım', 'Aralık'];

export default function MatchInfoCard({ m }) {
  const info = m?.info || {};
  const league = m?.league || '';
  const flag = countryFlag(info.country) || '';
  const leagueShort = info.country && league.startsWith(info.country) ? league.slice(info.country.length).trim() : league;
  const leagueText = [leagueShort, info.leagueWeek ? `${info.leagueWeek}. Hafta` : null].filter(Boolean).join(' · ');

  const d = m?.date ? new Date(m.date) : null;
  const valid = d && !Number.isNaN(d.getTime());
  const p = (n) => String(n).padStart(2, '0');
  const dateText = valid ? `${d.getDate()} ${AY[d.getMonth()]} ${d.getFullYear()}, ${WEEKDAY[d.getDay()]} · ${p(d.getHours())}:${p(d.getMinutes())}` : '';

  const rows = [];
  if (leagueText) rows.push({ icon: flag || '🏆', text: leagueText, strong: true });
  if (dateText) rows.push({ icon: '🕐', text: dateText });
  if (info.weather) rows.push({ icon: info.weather.emoji || '🌡️', text: `${info.weather.label} · ${info.weather.tempC}°C` });
  if (info.stadium) rows.push({ icon: '🏟️', text: info.city ? `${info.stadium}, ${info.city}` : info.stadium });

  if (!rows.length) return null;

  return (
    <View style={s.card}>
      {rows.map((r, i) => (
        <View key={i} style={[s.row, i > 0 && s.rowBorder]}>
          <Text style={s.icon}>{r.icon}</Text>
          <Text style={[s.text, r.strong && s.textStrong]} numberOfLines={1}>{r.text}</Text>
        </View>
      ))}
    </View>
  );
}

const s = StyleSheet.create({
  card: { backgroundColor: colors.card, borderRadius: radius.md, borderWidth: 1, borderColor: colors.border, paddingHorizontal: spacing.md, marginBottom: spacing.md, ...shadow.soft },
  row: { flexDirection: 'row', alignItems: 'center', gap: 10, paddingVertical: 10 },
  rowBorder: { borderTopWidth: 1, borderTopColor: colors.border },
  icon: { fontSize: 15, width: 22, textAlign: 'center' },
  text: { flex: 1, color: colors.textSoft, fontSize: 12.5, fontWeight: '700' },
  textStrong: { color: colors.text, fontSize: 13, fontWeight: '800' },
});

File: lib/features/match_detail/match_info_card.dart
// KAYNAK: app/src/components/MatchInfoCard.js — BİREBİR çeviri.
//
// Maç bilgi kartı (detay/Özet): bayrak + lig + lig haftası, tarih·gün·saat,
// hava, stadyum. Yalnız GERÇEK veri gösterilir; olmayan satır hiç çizilmez
// (uydurma yok). Veri kaynağı: backend m.info (+ m.league, m.date).

import 'package:flutter/material.dart';

import '../../core/theme/tokens.dart';
import '../../core/utils.dart';

const List<String> _weekday = [
  'Pazar',
  'Pazartesi',
  'Salı',
  'Çarşamba',
  'Perşembe',
  'Cuma',
  'Cumartesi',
];
const List<String> _ay = [
  'Ocak',
  'Şubat',
  'Mart',
  'Nisan',
  'Mayıs',
  'Haziran',
  'Temmuz',
  'Ağustos',
  'Eylül',
  'Ekim',
  'Kasım',
  'Aralık',
];

class MatchInfoCard extends StatelessWidget {
  const MatchInfoCard({super.key, required this.m});

  final Map<String, dynamic>? m;

  @override
  Widget build(BuildContext context) {
    final info = (m?['info'] as Map?) ?? const {};
    final league = '${m?['league'] ?? ''}';
    final country = info['country'] as String?;
    final flag = countryFlag(country);
    final leagueShort = (country != null && league.startsWith(country))
        ? league.substring(country.length).trim()
        : league;
    final leagueText = [
      if (leagueShort.isNotEmpty) leagueShort,
      if (info['leagueWeek'] != null) '${info['leagueWeek']}. Hafta',
    ].join(' · ');

    final d = m?['date'] is String
        ? DateTime.tryParse(m!['date'] as String)?.toLocal()
        : null;
    String p(int n) => n.toString().padLeft(2, '0');
    final dateText = d == null
        ? ''
        : '${d.day} ${_ay[d.month - 1]} ${d.year}, '
              '${_weekday[d.weekday % 7]} · ${p(d.hour)}:${p(d.minute)}';

    final weather = info['weather'] as Map?;
    final stadium = info['stadium'] as String?;
    final city = info['city'] as String?;

    // İKON TİPİ BİLEREK `Object` (kullanıcı isteği, 2026-08-12):
    // - `IconData` → UYGULAMANIN KENDİ arayüz ikonu, temaya göre boyanır.
    // - `String`   → VERİDEN gelen görsel: ülke bayrağı ve hava durumu
    //   emojisi. Bunlar kendi gerçek renklerinde kalmalı ("ülke bayrakları ve
    //   maç verisine ait görseller bu değişikliğe dahil edilmesin"), o yüzden
    //   metin olarak çizilirler.
    final rows = <({Object icon, String text, bool strong})>[
      if (leagueText.isNotEmpty)
        (
          icon: flag.isNotEmpty ? flag : Icons.emoji_events_outlined,
          text: leagueText,
          strong: true,
        ),
      if (dateText.isNotEmpty)
        (icon: Icons.schedule, text: dateText, strong: false),
      if (weather != null)
        (
          icon: weather['emoji'] is String && '${weather['emoji']}'.isNotEmpty
              ? '${weather['emoji']}'
              : Icons.thermostat,
          text: '${weather['label']} · ${weather['tempC']}°C',
          strong: false,
        ),
      if (stadium != null && stadium.isNotEmpty)
        (
          icon: Icons.stadium_outlined,
          text: (city != null && city.isNotEmpty) ? '$stadium, $city' : stadium,
          strong: false,
        ),
    ];

    if (rows.isEmpty) return const SizedBox.shrink();

    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.md),
      padding: const EdgeInsets.symmetric(horizontal: Spacing.md),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: AppRadius.mdR,
        border: Border.all(color: AppColors.border),
        boxShadow: AppShadow.soft,
      ),
      child: Column(
        children: [
          for (var i = 0; i < rows.length; i++)
            Container(
              padding: const EdgeInsets.symmetric(vertical: 10),
              decoration: i > 0
                  ? BoxDecoration(
                      border: Border(top: BorderSide(color: AppColors.border)),
                    )
                  : null,
              child: Row(
                children: [
                  SizedBox(
                    width: 22,
                    child: switch (rows[i].icon) {
                      final IconData d => Icon(
                        d,
                        size: 17,
                        color: AppColors.textSoft,
                      ),
                      // Bayrak / hava durumu — veriden gelir, kendi rengiyle.
                      final Object o => Text(
                        '$o',
                        textAlign: TextAlign.center,
                        style: const TextStyle(fontSize: 15),
                      ),
                    },
                  ),
                  const SizedBox(width: 10),
                  Expanded(
                    child: Text(
                      rows[i].text,
                      maxLines: 1,
                      overflow: TextOverflow.ellipsis,
                      style: TextStyle(
                        color: rows[i].strong
                            ? AppColors.text
                            : AppColors.textSoft,
                        fontSize: rows[i].strong ? 13 : 12.5,
                        fontWeight: rows[i].strong
                            ? AppFont.heavy
                            : AppFont.bold,
                      ),
                    ),
                  ),
                ],
              ),
            ),
        ],
      ),
    );
  }
}
